File: files/2026-07-19/CSEC 19072026.R
library(tidyverse)
library(readxl)

path <- "2026-07-19/Challenge 400.xlsx"
input <- read_excel(path, range = "B3:C12")
test <- read_excel(path, range = "E3:F6") %>%
  mutate(`Stock Out Days` = as.Date(`Stock Out Days`))

zero_runs <- rle(input$`Stock Bal` == 0)
ends <- cumsum(zero_runs$lengths)
starts <- c(1, head(ends, -1) + 1)

result <- tibble(
  `Stock Out Days` = as.Date(input$Date[starts[zero_runs$values]]),
  `Days Out` = zero_runs$lengths[zero_runs$values]
)

all.equal(result, test)
# TRUE

File: files/2026-07-19/CSEC 19072026.py
import pandas as pd

path = "2026-07-19/Challenge 400.xlsx"
input = pd.read_excel(path, usecols="B:C", nrows=9, skiprows=2)
test = pd.read_excel(path, usecols="E:F", nrows=3, skiprows=2)
test.columns = ["Stock Out Days", "Days Out"]

stock_is_zero = input["Stock Bal"].eq(0)
run_id = stock_is_zero.ne(stock_is_zero.shift()).cumsum()

rows = []
for _, run in input[stock_is_zero].groupby(run_id[stock_is_zero]):
    rows.append([run["Date"].iloc[0], len(run)])

result = pd.DataFrame(rows, columns=["Stock Out Days", "Days Out"])
print(result.equals(test))
